####Introduction
#Through the use of Salmonella enterica raw reads from an Oxford Nanopore sequencer (expected accuracy Q20+, N50:5-15kb), this work aims to assemble a consensus genome of high quality through read processing and long read assembly methods. The generated genome would then be compared to a reference genome from NCBI for Salmonella enterica that could be visualized. 
#Genome assembly can pose challenges such as ensuring accuracy with genome construction, especially when further analysis is required. Long-read sequences tend to have higher error rates in comparison to short read sequences, although recent improvements have led to more accurate results with long-read data alone (Zhao et al., 2023). Computational demands may also slow the progress of genome assembly and further analysis, so a pipeline with optimal balanced tools should be considered.  
A quality control tool was chosen to identify the condition of the raw Nanopore reads before further assembly and analysis were conducted. NanoPack was chosen due to its specific purpose of processing Nanopore reads, along with the inclusion of Nanoplot for data analysis (De Coster, 2018). A recent study looking at the use of Oxford nanopore long-read sequencing for genome assembly found that for a 5 Mb bacterial genome, accuracies of over 99.99% were achieved through the usage of Flye and Medaka for assembly and polishing (Zhao et al., 2023). Another study looking at the Salmonella enterica genome across 341 strains found that the average lengths of genomes ranged from 4.50 to 5.15 Mb, within the range of 5 Mb for high accuracy scores through Flye and Medaka (Lyu et al., 2021). Therefore, the combination of Flye and Medaka was considered for this project. Other assembly tools, such as Canu and NECAT also deliver accurate results in comparison studies however can lead to slower processing times (Kumar et al., 2025). Minimap2 was chosen as the aligner for this project as it is considered to be high performance for Nanopore reads, although studies have identified potential improvements for speed (Sadasivan, 2023). 


####Proposed Methods

#The data collected from NCBI contained Nanopore long-reads in the FASTQ format. Further analysis was conducted on the Compute Canada Fir cluster. A complete Salmonella enterica reference genome was retrieved from the NCBI database and used for alignment. Assembly and analysis tools were installed through the miniconda (v. 25.11.1) package manager through the Ubuntu terminal.
#Raw Nanopore reads were assessed for quality using NanoPlot (v. 1.46.2) with parameters --fastq and --format png (De Coster et al., 2018). Nanopack (v. 1.1.1) was used for further read edits as necessary (De Coster et al., 2018). Genome assembly was performed through the use of Flye (v. 2.9.6) (Kolmogorov et al., 2019). Polishing was done through the use of Medaka (v. 2.1.1) (Oxford Nanopore Technologies, 2025). In order to align the assembled genome to the reference, minimap2 was used with the -ax flag (v. 2.30), the alignments were then converted to BAM using samtools (v. 1.23) (Li, 2018);(Li et al., 2009). Aligned reads were visualized in Integrative Genomics Viewer (IGV) (v. 2.19.7) (Robinson et al., 2011) through the transfer of files locally. 


